### PR TITLE
feat:  Add  AudioWidget

### LINKED ISSUE #7062: [Feature] Create an Audio Player widget
## Summary

Create an Audio player widget to play Audio URLs

## Motivation

We are working on an Audio Recorder widget. So, users might also be looking for a widget to play the Audio

### PR DESCRIPTION
**Pull request by aswathkk**
## Description

Adds a AudioWidget by copying VideoWidget

Fixes #7062 

## Type of change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

## How Has This Been Tested?

Manually tested by using some random mp3 urls.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: feature/audio-widget 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 54.91 **(0.01)** | 37.03 **(-0.01)** | 33.91 **(0.02)** | 55.53 **(0.02)**
 :red_circle: | app/client/src/constants/messages.ts | 74.46 **(-0.02)** | 100 **(0)** | 23.83 **(-0.06)** | 80.86 **(0.04)**
 :red_circle: | app/client/src/selectors/commentsSelectors.ts | 78.23 **(-1.61)** | 54.41 **(-2.94)** | 70 **(0)** | 84.09 **(-2.27)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 50.47 **(-0.24)** | 38.26 **(-0.87)** | 36.21 **(0)** | 55.08 **(-0.27)**
 :sparkles: :new: | **app/client/src/widgets/AudioWidget/index.tsx** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/widgets/AudioWidget/widget/index.tsx** | **63.41** | **40** | **46.15** | **80.65**</details>

### CODE REVIEW

**Review by somangshu — APPROVED** [review]
Approving to run all tests

**Review by somangshu — DISMISSED** [review]
Approving to run tests

**Review by jsartisan — DISMISSED** [review]
LGTM!

**Comment by somangshu** [comment]
@aswathkk resolve conflicts and push to needs QA

**Comment by aswathkk** [comment]
@somangshu Can you approve so that the tests can run?

**Review by somangshu — APPROVED** [review]

**Comment by somangshu** [comment]
@aswathkk we need Jest and Cypress tests to cover the basic scenarios here

File: app/client/src/widgets/AudioWidget/widget/index.test.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import AudioWidget from ".";
+
+const urlTests = [
+  { url: "https://appsmith.com/", isValid: true },
+  { url: "http://appsmith.com/", isValid: true },
+  { url: "appsmith.com/", isValid: true },
+  { url: "appsmith.com", isValid: true },
+  { url: "release.appsmith.com", isValid: true },
+  { url: "appsmith.com/audio.mp3", isValid: true },
+  { url: "appsmith.com/audio.mp3", isValid: true },
```
Comment (somangshu) [inline app/client/src/widgets/AudioWidget/widget/index.test.tsx:11]: is this required, Can we remove the number of positive test cases here, I feel some are just repeated

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/DisplayWidgets/audio_spec.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+const widgetsPage = require("../../../../locators/Widgets.json");
+const commonlocators = require("../../../../locators/commonlocators.json");
+const dsl = require("../../../../fixtures/audioWidgetDsl.json");
+const testdata = require("../../../../fixtures/testdata.json");
+
+describe("Video Widget Functionality", function() {
```
Comment (somangshu) [inline app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/DisplayWidgets/audio_spec.js:6]: Audio widget functionality

**Review by somangshu — APPROVED** [review]
Looks good, Approving for tests to run

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/DisplayWidgets/audio_spec.js (hunk 2 above)
Comment (aswathkk) [reply app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/DisplayWidgets/audio_spec.js:6]: ok. I didn't notice that! 😉

File: app/client/src/widgets/AudioWidget/widget/index.test.tsx (hunk 1 above)
Comment (aswathkk) [reply app/client/src/widgets/AudioWidget/widget/index.test.tsx:11]: Yes. Should I just delete the repeated one or do you think most of them are unnecessary?

Comment (somangshu) [reply app/client/src/widgets/AudioWidget/widget/index.test.tsx:11]: Lets remove at-least the repeated once

**Review by somangshu — APPROVED** [review]
LGTM

**Comment by somangshu** [comment]
@aswathkk have we created documentation for this new widget, We should create a basic doc for this is not already done

**Comment by aswathkk** [comment]
@somangshu Documentation is being done on [#194](https://github.com/appsmithorg/appsmith-docs/pull/194). I will modify it and get it merged.

**Pull request merged**